src/Beberlei/Bundle/MetricsBundle/DependencyInjection/BeberleiMetricsExtension.php

```diff
@@ -18,31 +19,79 @@ public function load(array $configs, ContainerBuilder $container)
         $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
         $loader->load('metrics.xml');
 
-        $registry = $container->getDefinition('beberlei_metrics.registry');
-        $registry->addMethodCall('setDefaultName', array($config['default']));
+        foreach ($config['collectors'] as $name => $colConfig) {
+            $definition = $this->createCollector($colConfig['type'], $colConfig);
+            $container->setDefinition('beberlei_metrics.collector.'.$name, $definition);
+        }
+
+        if (!$config['default'] && 1 === count($config['collectors'])) {
+            $container->setAlias('beberlei_metrics.collector', 'beberlei_metrics.collector.'.$name);
+        } elseif ($container->hasDefinition('beberlei_metrics.collector.'.$config['default'])) {
+            $container->setAlias('beberlei_metrics.collector', 'beberlei_metrics.collector.'.$config['default']);
+        } else {
+            throw new \LogicException('You should select a default collector');
+        }
+    }
+
+    private function createCollector($type, $config)
+    {
+        $definition = new DefinitionDecorator('beberlei_metrics.collector._proto.'.$config['type']);
```

Comment:
If someone uses ``_proto.logger`` as name, they will be able to break service definitions. It would keep the proto on something which cannot conflict with collector services